### PR TITLE
feat(native): add native settings section

### PR DESCRIPTION
**Pull request by marekvospel**
This pull request adds a settings section that is only shown in elk native app and `minimize_to_tray` configuration option.

## Related
https://github.com/elk-zone/elk-native/pull/58

### CODE REVIEW

**Comment by patak-dev** [comment]
I wonder if we should add these options in elk-native itself, we should be able to add some hooks to extend the options 🤔 
cc @danielroe @JonasKruckenberg

**Comment by danielroe** [comment]
Yes. We could potentially move the tauri module that exists in this repo -> elk native repo itself.

**Comment by JonasKruckenberg** [comment]
I don't know how we would integrate this with the current submodule setup, but if you have any smart ideas, moving as much elk-native specific code into that repo sounds like a good solution to me generally

**Comment by marekvospel** [comment]
I don't think it would be possible to move the tauri module into elk-native repository, as that could cause some type issues (just adding tauri-runtime-only types in this repository I had problems with types, see https://github.com/elk-zone/elk/blob/8bd02816203f5e00c7440689a877a5888107613a/modules/tauri/runtime/native-settings-undefined.ts)

At least I can't think of any good and easy solution

**Comment by marekvospel** [comment]
I've resolved the merge conflicts. Should we wait till https://github.com/elk-zone/elk-native/issues/65 is resolved before merging this?

I feel like for now we could just merge it and once it is decided whether we will continue to use git submodules or start using other methods, just migrate some of this code. I'll gladly migrate this into elk-native once and in case it is necessary.

**Comment by marekvospel** [comment]
@JonasKruckenberg This method could be added to `tauri-plugin-store` to make it easier to work with the values reactively. However it is framework specific, so there could be some problems with dependencies and other framework integrations would probably need to be added.
https://github.com/elk-zone/elk/blob/c383d93989224b519fdb826e94d16c19c3d6fa2a/modules/tauri/runtime/native-settings.ts#L8-L35

**Comment by marekvospel** [comment]
@Shinigami92 any chance for a review? This PR has been sitting here without any activity for a while

File: modules/tauri/runtime/native-settings.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        await store.set(key, value)
+        await store.save()
+        refValue = value
+        trigger()
+      },
+    }
+  })
+
+  const result = await store.get(key)
+  if (result !== undefined && result !== null)
```
Comment (Shinigami92) [inline modules/tauri/runtime/native-settings.ts:31]:
```suggestion
  if (result != null)
```

File: modules/tauri/runtime/native-settings.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    refValue = result as T
+
+  return ref
+}
+
+async function nativeSettings() {
```
Comment (Shinigami92) [inline modules/tauri/runtime/native-settings.ts:37]:
```suggestion
async function nativeSettings(): Promise<NativeSettings> {
```

File: modules/tauri/runtime/native-settings.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  return {
+    minimizeToTray: await tauriStoreRef(store, 'minimize_to_tray', true),
+  }
+}
+
+interface NativeSettings {
```
Comment (Shinigami92) [inline modules/tauri/runtime/native-settings.ts:43]:
```suggestion
export interface NativeSettings {
```

File: modules/tauri/runtime/native-settings.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  minimizeToTray: TauriStoreRef<boolean>
+}
+export type { NativeSettings }
```
Comment (Shinigami92) [inline modules/tauri/runtime/native-settings.ts:46]:
```suggestion
```

File: modules/tauri/runtime/native-settings.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+async function nativeSettings() {
+  return {
+    minimizeToTray: await tauriStoreRef(store, 'minimize_to_tray', true),
+  }
+}
+
+interface NativeSettings {
+  minimizeToTray: TauriStoreRef<boolean>
+}
```
Comment (Shinigami92) [inline modules/tauri/runtime/native-settings.ts:45]: IMO you should move this above `nativeSettings()`

File: modules/tauri/runtime/native-settings.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+export type { NativeSettings }
+export const useNativeSettings = async () => await nativeSettings() as NativeSettings | undefined
```
Comment (Shinigami92) [inline modules/tauri/runtime/native-settings.ts:47]:
```suggestion
export { useNativeSettings: nativeSettings };
```

not sure if `:` or `as`

**Review by Shinigami92 — COMMENTED** [review]
I could just do a code review
I don't use the native tauri app, you should ask someone else that is more familiar with the tauri app

**Comment by marekvospel** [comment]
> I could just do a code review I don't use the native tauri app, you should ask someone else that is more familiar with the tauri app

@JonasKruckenberg could you please take a look at this when you're free?

File: modules/tauri/index.ts [hunk 7]
```diff
@@ -9,8 +9,10 @@ export default defineNuxtModule({
     const nuxt = useNuxt()
     const { resolve } = createResolver(import.meta.url)
 
-    if (!process.env.TAURI_PLATFORM)
+    if (!process.env.TAURI_PLATFORM) {
+      addImports({ name: 'useNativeSettings', from: resolve('./runtime/native-settings-undefined') })
```
Comment (Shinigami92) [inline modules/tauri/index.ts:13]: Could you please explain why it is needed to have this `-undefined` version?
Cant this be e.g. inlined like `addImports({ name: 'useNativeSettings', from: () => undefined })` or similar?

File: modules/tauri/runtime/native-settings.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        await store.set(key, value)
+        await store.save()
+        refValue = value
+        trigger()
+      },
+    }
+  })
+
+  const result = await store.get(key)
+  if (result !== null)
```
Comment (Shinigami92) [inline modules/tauri/runtime/native-settings.ts:31]: is it correct here that undefined is not covered?

File: modules/tauri/index.ts (hunk 7 above)
Comment (marekvospel) [reply modules/tauri/index.ts:13]: I didn't find any way to add an import inline, the import object requires `from: string` value, and I couldn't find any other auto import releated function, that would do such a thing.

**Comment by marekvospel** [comment]
Closing, as elk-native is no longer actively developed.

**Pull request closed without merging**